assets/js/googlesitekit/datastore/user/adblocker.js

```diff
@@ -0,0 +1,146 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { detectAnyAdblocker } from 'just-detect-adblock';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from './constants';
+
+// Actions
+const CHECK_ADBLOCKER = 'CHECK_ADBLOCKER';
+const RECEIVE_IS_ADBLOCKER_ACTIVE = 'RECEIVE_IS_ADBLOCKER_ACTIVE';
+
+export const initialState = {
+	isAdBlockerActive: undefined,
+};
+
+export const actions = {
+	*checkAdBlocker() {
+		return yield {
+			payload: {},
+			type: CHECK_ADBLOCKER,
+		};
+	},
+	receiveIsAdBlockerActive( isAdBlockerActive ) {
+		invariant(
+			'boolean' === typeof isAdBlockerActive,
+			'isAdBlockerActive must be boolean.'
+		);
+		return {
+			payload: { isAdBlockerActive },
+			type: RECEIVE_IS_ADBLOCKER_ACTIVE,
+		};
+	},
+};
+
+export const controls = {
+	[ CHECK_ADBLOCKER ]: async () => {
+		if ( await detectAnyAdblocker() ) {
+			return true;
+		}
+		// The above is good about detecting most adblockers.
+		// For the rest, we'll make a placeholder request to the favicon with some
+		// additional stuff in the query string to (hopefully) trigger a filter.
+		// If this throws, then the fetch request failed completely and we'll assume it was blocked.
+		try {
+			const params = [
+				// The name of the parameter here doesn't really matter
+				// since adblockers look at the URL as a whole.
+				// Note: this value must not be URL-encoded.
+				'google-site-kit=/adsense/pagead2.googlesyndication.com/pagead/js/adsbygoogle.js',
+				// Add a timestamp for cache-busting.
+				`timestamp=${ Date.now() }`,
+			];
+			await fetch( `/favicon.ico?${ params.join( '&' ) }`, {
+				credentials: 'omit',
+				// Don't follow any redirects; we only care about this request being blocked or not.
+				redirect: 'manual',
+			} );
+		} catch {
+			return true;
+		}
+		return false;
+	},
+};
+
+export const reducer = ( state, { payload, type } ) => {
+	switch ( type ) {
+		case RECEIVE_IS_ADBLOCKER_ACTIVE: {
+			const { isAdBlockerActive } = payload;
+
+			return {
+				...state,
+				isAdBlockerActive,
+			};
+		}
+
+		default: {
+			return state;
+		}
+	}
+};
+
+export const resolvers = {
+	*isAdBlockerActive() {
+		const registry = yield Data.commonActions.getRegistry();
+		const isAdBlockerActive = registry
+			.select( CORE_USER )
+			.isAdBlockerActive();
+
+		// If ad blocker status was already detected, consider it fulfilled
+		// and don't check the global.
```

Comment:
I suspect "checking the global" is outdated here regarding what we check, given this [original commit is four years old](https://github.com/google/site-kit-wp/commit/f4263c27379eb19212e057fc9c631dd448e6ed54) 😅 

```suggestion
		// If the ad blocker status is already set, don't make any requests
		// to check the ad blocker status.
```